[lldb/API] Fix SBStructuredData support any JSON type

medismailben opened the pull request:
This patch loosen the parsing requirement to allow parsing not only
JSON dictionaries but also valid JSON type (integer, float, string,
bool, array, null).

Signed-off-by: Med Ismail Bennani <ismail@bennani.ma>

DavidSpickett reviewed (approved):
LGTM.

labath commented on lldb/test/API/python_api/sbstructureddata/TestStructuredDataAPI.py:
```diff
@@ -110,6 +110,23 @@ class MyRandomClass:
+
+        # Check SetFromJSON API for dictionaries, integers, floating point
+        # values, strings and arrays
+        error = example.SetFromJSON(s)
+        if not error.Success():
+            self.fail("FAILED:   " + error.GetCString())
```
```suggestion
        self.assertSuccess(example.SetFromJSON(s))
```

labath replied:
I see you've resolved this comment, but I don't actually understand how. Is your code in some way better than assertSuccess (which exists precisely so it can print the error message when the SBError is in failure state)?

labath replied:
Ah, maybe I see... You've changed one instance to assertSucess, but then added another one with the old pattern, right?

If so, then just change the other one as well, and this is good to go :)

labath reviewed (commented):
How is the operator bool change related to this patch? AFAICT, the pimpl pointer should always be valid.

bulbazord commented on lldb/source/API/SBStructuredData.cpp:
```diff
@@ -106,7 +110,7 @@ bool SBStructuredData::IsValid() const {
   LLDB_INSTRUMENT_VA(this);
 
-  return m_impl_up->IsValid();
+  return m_impl_up && m_impl_up->IsValid();
```
I'm not sure how this is related to this patch, but it shouldn't be necessary. `SBStructuredData` should always have a valid opaque pointer.

medismailben commented:
@labath, does this looks good to you ?

clayborg commented:
Shouldn't any valid JSON be acceptable? A dictionary, array, string, number, "true", "false", or "null"?

medismailben commented:
> Shouldn't any valid JSON be acceptable? A dictionary, array, string, number, "true", "false", or "null"?

Sure, fixed that @clayborg  :)

medismailben committed:
[lldb/API] Fix SBStructuredData support any JSON type

This patch loosen the parsing requirement to allow parsing not only
JSON dictionaries but also valid JSON type (integer, float, string,
bool, array, null).

Signed-off-by: Med Ismail Bennani <ismail@bennani.ma>
